packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -353,15 +629,235 @@ fn expand_input_param(
     }
 }
 
+// @todo This is an experimental support for the new JSON ABI file format.
+// Once this is stable:
+// 1. Delete old one;
+// 2. Rename it to its original name;
+// 3. Write documentation.
+fn _new_expand_input_param(
+    fun: &ABIFunction,
+    type_application: &TypeApplication,
+    kind: &ParamType,
+    types: &HashMap<usize, TypeDeclaration>,
+) -> Result<TokenStream, Error> {
+    match kind {
+        ParamType::Array(ty, _) => {
+            let ty = _new_expand_input_param(fun, type_application, ty, types)?;
+            Ok(quote! {
+                ::std::vec::Vec<#ty>
+            })
+        }
+        ParamType::Enum(_) => {
+            let t = types.get(&type_application.type_field).unwrap(); // temp unwrap
+
+            let ident = ident(&_new_extract_custom_type_name_from_abi_property(
+                t,
+                Some(CustomType::Enum),
+                types,
+            )?);
+            Ok(quote! { #ident })
+        }
+        ParamType::Struct(_) => {
+            let t = types.get(&type_application.type_field).unwrap(); // temp unwrap
+
+            let ident = ident(&_new_extract_custom_type_name_from_abi_property(
+                t,
+                Some(CustomType::Struct),
+                types,
+            )?);
+            Ok(quote! { #ident })
+        }
+        // Primitive type
+        _ => expand_type(kind),
+    }
+}
+
 // Regarding string->TokenStream->string, refer to `custom_types_gen` tests for more details.
 #[cfg(test)]
 mod tests {
+    use fuels_types::ProgramABI;
+
     use crate::EnumVariants;
     use std::slice;
 
     use super::*;
     use std::str::FromStr;
 
+    #[test]
+    fn test_expand_function_simple_new_abi() -> Result<(), Error> {
+        let s = r#"
+        {
+            "types": [
+              {
+                "typeId": 6,
+                "type": "u64",
+                "components": null,
+                "typeParameters": null
+              },
+              {
+                "typeId": 8,
+                "type": "b256",
+                "components": null,
+                "typeParameters": null
+              },
+              {
+                "typeId": 6,
+                "type": "u64",
+                "components": null,
+                "typeParameters": null
```

Comment:
same as above. Should we have it duplicated ?

Reply:
Answered in https://github.com/FuelLabs/fuels-rs/pull/533#discussion_r954248751.